service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/CourtServiceChecker.java

```diff
@@ -27,15 +28,44 @@ public boolean isStarted(CaseData caseData) {
         }
 
         return anyNonEmpty(hearingPreferences.getWelsh(),
-                hearingPreferences.getInterpreter(),
-                hearingPreferences.getIntermediary(),
-                hearingPreferences.getDisabilityAssistance(),
-                hearingPreferences.getExtraSecurityMeasures(),
-                hearingPreferences.getSomethingElse());
+            hearingPreferences.getInterpreter(),
+            hearingPreferences.getIntermediary(),
+            hearingPreferences.getDisabilityAssistance(),
+            hearingPreferences.getExtraSecurityMeasures(),
+            hearingPreferences.getSomethingElse());
     }
 
     @Override
     public boolean isCompleted(CaseData caseData) {
-        return false;
+        final HearingPreferences hearingPreferences = caseData.getHearingPreferences();
+
+        if (hearingPreferences == null || anyEmpty(hearingPreferences.getWelsh(),
+            hearingPreferences.getInterpreter(),
+            hearingPreferences.getIntermediary(),
+            hearingPreferences.getDisabilityAssistance(),
+            hearingPreferences.getExtraSecurityMeasures(),
+            hearingPreferences.getSomethingElse())) {
+            return false;
+        }
+
+        if (("Yes").equals(hearingPreferences.getWelsh())
+            && isEmpty(hearingPreferences.getWelshDetails())) {
+            return false;
+        } else if (("Yes").equals(hearingPreferences.getInterpreter())
```

Comment:
We have enum YesOrNo.java that could maybe be used here and other places instead of the string "Yes".. we use it like YES.getValue()..